libs/marketplace/src/fetcher.ts

```diff
@@ -31,7 +33,7 @@ export const listPositons = async ({
     .filter((l) => !owner || l.to.toLocaleLowerCase() === owner.toLocaleLowerCase())
   const { stakingV1ToStakingPosition } = parser(stakingV1Contract, provider)
   const result = await Promise.all(preFilter.map(stakingV1ToStakingPosition))
-  return result.filter((p) => !p.currentValue.eq(0)) //remove redeemeds
+  return result.filter((p) => excludeRedeemed && !p.currentValue.eq(0)) //remove redeemeds
```

Comment:
no need to filter if `excludeRedeemed ` is false
```suggestion
  return excludeRedeemed ? result.filter((p) => p.currentValue.eq(0)) : result
```